pauthabielf64/pauthabielf64.rst

```diff
@@ -861,198 +740,313 @@ main can. The static linker must communicate the details of how to
-code and implementation of dlsym agree on. A simple implicit signing
-schema is for dlsym to sign code pointers but not data pointers. A
-more complex implemenation could add metadata to inform the dynamic
-linker how dlsym should sign the pointer. Finally variants of the
-dlsym, and dlvsym functions could be added with an extra parameter for
-the signing schema.
-
-TBD A mechanism to support different signing schemas for ``dlsym`` and
-``dlvsym``.
-
-Questions/Issues
+code and implementation of ``dlsym`` agree on.
 
-* Feedback on whether a signing schema is needed for runtime dynamic linking.
+The PAuth ABI uses a simple simple default signing schema. If the
+symbol found by ``dlsym`` has type ``STT_FUNC`` the address to be
+returned is signed with the ``IA`` key with a 0 modifier. Otherwise
+the address is not signed.
 
+An optional extension that communicates the signing schema for a
+symbol to the dynamic linker can be found in Appendix extension to
+recording signing schema for ``dlsym``.
 
 ELF Marking
 ===========
 
-As an experimental ABI, marking ELF files that use this ABI is
-optional; it is the experimenter's responsibility to match compatible
-relocatable object files and link-units. Once this specification is
-used in production ELF files must be marked to allow toolchains and
-platforms to reason about compatibility. In contrast to much of the
-ABI the high-level language mapping of source language to signing
-schema is expected to evolve over time. Even if the low-level ELF
-extensions remain constant a change to the high-level language mapping
-will result in incompatible ELF files.
-
-Prior Art
----------
+ELF files must be marked to allow toolchains and platforms to reason
+about compatibility. The high-level language mapping of source
+language to signing schema is expected to evolve over time. Even if
+the low-level ELF extensions remain constant, a change to the
+high-level language mapping may result in incompatible ELF files.
 
-* AAELF64_ defines the .note.gnu.property
-  ``GNU_PROPERTY_AARCH64_FEATURE_1_AND`` with a feature bit
-  ``GNU_PROPERTY_AARCH64_FEATURE_1_PAC`` which indicates that all
-  executable sections have Return Address Signing enabled.
+Every relocatable object, executable and shared library that uses the
+PAuth ABI ELF extensions must have a section named
+``.note.AARCH64-PAUTH-ABI-tag`` of type ``SHT_NOTE``. This section is
+structured as a note section as documented in SCO-ELF_.
 
-* AAELF64_ defines a dynamic tag ``DT_AARCH64_PAC_PLT`` that a static
-  linker must produce if the PLT sequences expect the .plt.got entries
-  to be signed by the dynamic linker.
+The name field (``namesz`` / ``name``) contains the string "ARM". The
+type field shall be 1. The ``descsz`` field must be at least 16, with
+the first 16 bytes of the description containing 2 64-bit words. With
+the first 64-bit word a platform identifier, and the second 64-bit
```

Comment:
.. 64-bit word `being` a platform ..

.. second 64-bit identifier `being` a version ..

Reply:
I've reworded a bit to make it clearer that the second word is a version number for the platform given by the first.